test/com/winterwell/es/client/TransformRequestBuilderDatehistogramTest.java

```diff
@@ -0,0 +1,106 @@
+	
+	private static final String P_TIME = "time";
+	private static final String P_AGG = "agg";
+	private static final String P_COUNT = "count";	
+	
+	@Before
+	public void setup()
+	{
+		Dep.setIfAbsent(ESConfig.class, new ESConfig());
+		if ( ! Dep.has(ESHttpClient.class)) Dep.setSupplier(ESHttpClient.class, false, ESHttpClient::new);
+		IndicesAdminClient indicesClient = Dep.get(ESHttpClient.class).admin().indices();
+		
+		try {									// Scrub any existing version of the source index
+			indicesClient.prepareDelete(SRC).get().check();
+		} catch (ESIndexNotFoundException e) { /* Ignore */ }
+
+		try {									// Scrub any existing version of the destination index
+			indicesClient.prepareDelete(DEST).get().check();
+		} catch (ESIndexNotFoundException e) { /* Ignore */ }
+		
+												// Create the source
+		indicesClient.prepareCreate(SRC).get().check();
+		
+		indicesClient.preparePutMapping(SRC)	// Create the source property map
+				.setMapping(new ESType()
+						.property(P_TIME, new ESType().date())
+						.property(P_AGG, new ESType().LONG())
+						.property(P_COUNT, new ESType(double.class))
+				).get().check();
+	}
+	
+	private static final int HOUR_MS = 1000*60*60;
+
+	@Test
+	public void oneHour() {
+		try {
+
+			ESHttpClient esc = Dep.get(ESHttpClient.class);
+	
+			Time baseTs = new Time("2022-11-01T12:00:00Z");		// Midday UTC on the nose
+			
+			// Under the hood ES stores dates as ms since the epoch in UTC so our quanta is 1ms
+			
+			esc.prepareBulk()									// Insert rows into source index
+			.add(esc.prepareIndex(SRC, "id1").setBodyMap(new ArrayMap<String, Object>(
+					P_TIME, baseTs, P_AGG, 1, P_COUNT, 1)))
+			.add(esc.prepareIndex(SRC, "id2").setBodyMap(new ArrayMap<String, Object>(
+					P_TIME, baseTs.plus(1, TUnit.MILLISECOND), P_AGG, 10, P_COUNT, 1)))
+			.add(esc.prepareIndex(SRC, "id3").setBodyMap(new ArrayMap<String, Object>(
+					P_TIME, baseTs.plus(HOUR_MS - 1, TUnit.MILLISECOND), P_AGG, 100, P_COUNT, 1)))
+			.add(esc.prepareIndex(SRC, "id4").setBodyMap(new ArrayMap<String, Object>(
+					P_TIME, baseTs.plus(HOUR_MS, TUnit.MILLISECOND), P_AGG, 1000,  P_COUNT, 1)))
+			.add(esc.prepareIndex(SRC, "id5").setBodyMap(new ArrayMap<String, Object>(
+					P_TIME, baseTs.plus(HOUR_MS + 1, TUnit.MILLISECOND), P_AGG, 10000, P_COUNT, 1)))
+			.add(esc.prepareIndex(SRC, "id6").setBodyMap(new ArrayMap<String, Object>(
+					P_TIME, baseTs.plus(2*HOUR_MS - 1, TUnit.MILLISECOND), P_AGG, 100000, P_COUNT, 1)))
+			.add(esc.prepareIndex(SRC, "id7").setBodyMap(new ArrayMap<String, Object>(
+					P_TIME, baseTs.plus(2*HOUR_MS, TUnit.MILLISECOND), P_AGG, 1000000, P_COUNT, 1)))
+			.get().check();
+			
+			esc.transform().prepareTransform(JOB_ID)			// Create the transform
+			.setBody(SRC, DEST, Arrays.asList(P_AGG), Collections.emptyList(), "1h")
+			.get().check();
+																// Start the transform
+			esc.transform().prepareTransformStart(JOB_ID).get().check();
+			
```

Comment:
@winterstein the above seems to create a sensible dataset, and as I mentioned I can manually do a search on it with fixed_time aggregation, but this transform doesn't do anything so far. I end up with what looks like a sensible transform in status "stopped" with trigger count 1 and documents processed 0 and the destination index created but empty.

Reply:
Let's test the ComrpessImpressions.java code (which we know does stuff on the server - and seems to behave correctly apart from the oddity of creating docs an hour into the next day)